spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/util/FormUtils.java

```diff
@@ -955,29 +951,29 @@ private static void processLibraryQC(String libraryQc, String libraryQcMolarity,
       }
 
       try {
-        LibraryQC lqc = new LibraryQCImpl();
+        LibraryQC lqc = new LibraryQC();
         lqc.setLibrary(library);
         lqc.setResults(Double.valueOf(libraryQcMolarity));
-        lqc.setQcCreator(u.getLoginName());
-        lqc.setQcDate(new Date());
-        QcType lqct = libraryQcStore.getLibraryQcTypeByName(libraryQc);
+        lqc.setCreator(u);
+        lqc.setDate(new Date());
+        QcType lqct = qcService.getQcType("Library", libraryQc);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Library" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)

Reply:
:+1: